### DISCUSSION TITLE
Berry Script MQTT Publish format in console - cant see full message

### DISCUSSION THREAD

**Discussion by paulobriennz** [arendst/Tasmota#17666] [Q&A]
i am using berry script and publishing a MQTT message. in the console, it doesn't show the message normally. it shows it like this:

15:18:31.123 MQT: xxxxx-test31/SENSOR = 7B2254696D65223A2022323032332D [{"Time": "2023-0] ...

How do i see the whole message and whats the 7B22.... thing at the beginning of it??

**Comment by barbudor** [comment]
Without a crystal ball I have no idea what you have done wrong but everything is in the docs
mqtt.publish expect a string as payload argument
If you have an map object, you can encode it using the json.dump() method

https://tasmota.github.io/docs/Berry/#mqtt-module

https://github.com/berry-lang/berry/wiki/Chapter-7#dump-function

**Reply by sfromis** [reply]
The `7B22`.... is simply a hex representation of the start of the payload.

The whole message will not be visible in the Tasmota console, you can use a MQTT client to see data. One possible MQTT client is a single-liner in Berry itself, like 
`import mqtt mqtt.subscribe('xxxxx-test31/SENSOR',/topic,idx,data,databytes->print(topic,data))`

Of course, in case of you having the code doing the publish, you can also simply insert a `print(...)` at that point.

**Reply by paulobriennz** [reply]
thanks guys, the driver approach was the solution for me here, thanks for explaining the hex bit.
